manager/allocator/networkallocator/networkallocator.go

```diff
@@ -303,38 +324,84 @@ func OnInit(options *ServiceAllocationOpts) {
 // IsServiceAllocated returns if the passed service has its network resources allocated or not.
 // init bool indicates if the func is called during allocator initialization stage.
 func (na *NetworkAllocator) IsServiceAllocated(s *api.Service, flags ...func(*ServiceAllocationOpts)) bool {
-	var options ServiceAllocationOpts
+	return !na.IsServiceToBeAllocated(s, flags...)
+}
```

Comment:
Rather than define both the function and its inverse, let's choose one or the other.